pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -384,10 +414,18 @@ func (r *ReconcileStackConfigPolicy) updateStatus(ctx context.Context, scp polic
 }
 
 func (r *ReconcileStackConfigPolicy) onDelete(ctx context.Context, obj types.NamespacedName) error {
-	return resetOrphanSoftOwnedSecrets(ctx, r.Client, obj, nil)
+	return handleOrphanSoftOwnedSecrets(ctx, r.Client, obj, nil)
+}
+
+func handleOrphanSoftOwnedSecrets(ctx context.Context, c k8s.Client, softOwner types.NamespacedName, configuredEs esMap) error {
+	err := resetOrphanSoftOwnedSecrets(ctx, c, softOwner, configuredEs)
+	if err != nil {
+		return err
+	}
+	return deleteOrphanSoftOwnedSecrets(ctx, c, softOwner, configuredEs)
```

Comment:
Not sure if this already an issue with the existing implementation but secrets that were once referenced in the stack config policy but are no longer there will not be garbage collected unless the stack config policy itself is deleted.

Reply:
We discussed this offline, I am going to create a separate issue and address this in a separate PR